### PR TITLE
[Utilities] fix adding unsupported constraints to AbstractModel

### PR DESCRIPTION
**Pull request by odow**
Closes #2750
Cloess #2751

This PR is an alternative to #2751

### CODE REVIEW

File: src/Utilities/model.jl [hunk 1]
```diff
@@ -328,10 +335,17 @@ end
...
-    f::MOI.VariableIndex,
-    s::MOI.AbstractScalarSet,
-)
-    return MOI.add_constraint(model.variables, f, s)
+    func::MOI.VariableIndex,
+    set::S,
+) where {S<:MOI.AbstractScalarSet}
+    # We check supports_constraint here because it is a common practice for
+    # AbstractModels to declare that they do not support particular constraints,
+    # evenn though the underlying `.constraints` object does. See, for example,
```
Comment (blegat) [inline src/Utilities/model.jl:343]:
```suggestion
    # even though the underlying `.constraints` object does. See, for example,
```

**Comment by blegat** [comment]
Ah, I understand better your other PR now, it's not such a nasty bug. But I wouldn't say it was expected behavior that we were permissive by letting `add_constraint` work even when `supports_constraint` was defined. I think this PR gives the expected behavior.

**Review by blegat — APPROVED** [review]

**Comment by odow** [comment]
Yeah it's a case of "things worked when they shouldn't have." It also never came up in practice at the JuMP level, because we always copy to a file format with bridges, and they checked appropriately.

File: src/Utilities/model.jl [hunk 2]
```diff
@@ -312,9 +312,16 @@ end
...
     model::AbstractModel,
-    func::MOI.AbstractFunction,
-    set::MOI.AbstractSet,
-)
+    func::F,
+    set::S,
+) where {F<:MOI.AbstractFunction,S<:MOI.AbstractSet}
+    # We check supports_constraint here because it is a common practice for
+    # AbstractModels to declare that they do not support particular constraints,
+    # evenn though the underlying `.constraints` object does. See, for example,
```
Comment (odow) [inline src/Utilities/model.jl:320]:
```suggestion
    # even though the underlying `.constraints` object does. See, for example,
```

**Pull request merged**